### ISSUE TITLE
Error when retrieving a model save as .rds

### ISSUE THREAD

**Issue by DiegoPavonJordan** [glmmTMB/glmmTMB#394]
Dear all,
I, some time ago, run a model using glmmTMB and saved it as an .rsd object. Then, I could 'load' the model again using the readRDS command. So it worked well.
Now, when the revision time has come, I ran the same code but I am not able to get to the model results. When I type:
m1 <- readRDS('Model_unprotIBA2_log.rds')
summary(m1)

I get:

> summary(m1)
Error in if (REML) { : argument is of length zero

And if I when I ran:

> m1
Formula:          
LogAbu ~ Anomaly_std + NEnessSpp_std + Unprotected_IBA + Winter_std +  
    Unprotected_IBA * Winter_std + NEnessSpp_std * Unprotected_IBA +  
    Winter_std * NEnessSpp_std + Anomaly_std * NEnessSpp_std +  
    Winter_std * NEnessSpp_std * Unprotected_IBA + (1 | site) +      (1 | species)
Data: UnprotIBA
     AIC      BIC   logLik df.resid 
 6031107  6031274 -3015541  2707636 
Error in retape() : 
  INTEGER() can only be applied to a 'integer', not a 'NULL'

I understood from Molli that this is because I ran the models in an older version of glmmTMB, without the REML argument. Now, I have installed the latest version of glmmTMB (0.2.2.0), which is supposed to deal with this incompatibility (if I understood correctly from Ben), but I got the errors showed above. 

In other circumstances, I would not mind to install an older version of glmmTMB and run the models again, but these just take awfully a lot of time to run.

Do you know if there would be a way to retrieve the models from those óld´ .rds objects?

Thank you very much!

Diego

**Comment by strengejacke** [comment]
The problem is that the new version of glmmTMB stores additional information in the model-object, and the `summary()`-function of the latest glmmTMB-version tries to access this information - which is not stored in your model-object (because, as you correctly noticed, used an older version of glmmTMB).

So you either need to install an older version of glmmTMB to access your model-summary, or re-fit your model using the latest package version.

**Comment by bbolker** [comment]
- Reiterating from my previous comment,  your first error is surprising and should be fixable (what do you get if you call `traceback()` after getting the error? (For the record, I would have expected it to be fixed via https://github.com/glmmTMB/glmmTMB/commit/aa1a546e5d3e419a2aacd18bcaa2ef22a4a42f69 ).
- As for the "retape" error: this is a little less obvious. I have had problems with `glmmTMB` objects reloaded from file, but nothing that I've been able/had time to reproduce and track down. I assume there's something subtle about the (de)serialization of the binary objects that make up the internal TMB structures ...

I can imagine that your model object is huge, but if you'd like to post it somewhere I can try to take a look.

**Comment by DiegoPavonJordan** [comment]
Yes, thank you. I installed an earlier version of R and all packages and I could use summary() again. 
I just posted this here in case you can/want to do something about this issue. 

@bbolker running the traceback either crashes RStudio or takes forever to finish. If you think that the information from traceback could be useful for you and others, I can run it to the end and send you the report.

Thanks a lot!

**Comment by bbolker** [comment]
can you set `options(error=recover)` and report the output when the program stops?
